src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.CompilationAnalyzer.cs

```diff
@@ -64,8 +64,7 @@ private static void ReportDiagnostics(
             {
                 foreach (var diagnostic in diagnostics)
                 {
-                    if (locationFilter(diagnostic.Location) &&
-                        diagnostic.Severity != DiagnosticSeverity.Hidden)
+                    if (locationFilter(diagnostic.Location))
```

Comment:
this should be ok.  there are only a few hidden diagnostics, and having them call out to providers is reasonable.